### DISCUSSION TITLE
Hope this gets some love in 2023!

### DISCUSSION THREAD

**Discussion by Erudition** [ianmackenzie/elm-3d-scene#108] [General]
Loving what I can do with this library - I'm already able to build a lot of what I build in OpenSCAD, but with type safety. Nice!
It looks like there hasn't been any activity in 2 years, even some open pull requests, so I hope the project is not doomed to stagnate.

**Comment by ianmackenzie** [comment]
Not at all @Erudition! There was indeed a longish pause in the last couple years (due in large part to a new job, new partner, a marriage and now a baby) but I've been at work preparing for the new release which will include normal (bump) mapping and skybox textures. I actually recently published updates to a few dependency libraries including `elm-geometry` and `elm-3d-camera` with some changes that were needed for the new `elm-3d-scene` version 🙂 

I admittedly don't have quite as much free time for open-source work these days with a toddler around, but for what it's worth I'm still doing Elm full-time at work including working with `elm-3d-scene` so I'm still very much engaged with the Elm ecosystem!

**Reply by ianmackenzie** [reply]
BTW if you have any questions, or just fun screenshots to share, the **#geometry** and **#webgl** channels on the [Elm Slack](https://elm-lang.org/community/slack) are good 🙂 I tend to monitor **#geometry** more carefully myself but if you mention `elm-3d-scene` anywhere in Slack I'll get a notification 😅

**Comment by ianmackenzie** [comment]
Oh and I'd love to see what kind of geometry you've created! Geometry generation is very near and dear to my heart...

**Reply by Erudition** [reply]
Thanks for the replies! Good luck with those life changes <3

Personally I just started. I'm recreating [this diagram](https://nationalcpc.org/images/3D-cube.png) in 3D, and after forking an example I got this:
https://ellie-app.com/pnXY7W58qCLa1

I'll need to add the text legend, but after seeing the other Ellie example where the overlay is actually all done in SVG with a complicated 2D projection that I don't yet grok, and realizing the library doesn't natively support "what object is the mouse over", I have a lot of work cut out for me lol.

**Reply by ianmackenzie** [reply]
The cubes look good! But yeah for labels you'll definitely want to use overlaid SVG, I use that technique a lot. Happy to help with the 2D projection - agreed that it's tricky to conceptualize/grok but I think the various projection functions in [`elm-3d-camera`](https://package.elm-lang.org/packages/ianmackenzie/elm-3d-camera/latest/) at least make the necessary code pretty short.

As for "what object is the mouse over", you can do some of that using a combination of [`Camera3d.ray`](https://package.elm-lang.org/packages/ianmackenzie/elm-3d-camera/latest/Camera3d#ray) to get a "pick ray" corresponding to the currrent mouse position, and then the various [`Axis3d` intersection functions](https://package.elm-lang.org/packages/ianmackenzie/elm-geometry/latest/Axis3d#intersection) to figure out what objects that ray actually hits. Eventually it would be great to have `elm-3d-scene` support this more natively, but at least several of the trickiest low-level bits of functionality are already in place.
